app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+        {
+            // Defines the xml file for the fragment
+            return inflater.inflate(R.layout.activity_gallery_content, parent, false);
+        }
+
+        /**
+         * This event is triggered soon after onCreateView().
+         * Any view setup should occur here.
+         */
+        @Override
+        public void onViewCreated(View view, Bundle savedInstanceState)
+        {
+            //Create the container for all my cards
+            RecyclerView picturesRecycleView = view.findViewById(R.id.recycle_view);
+            picturesRecycleView.setHasFixedSize(true);
+
+            //Set a grid (2 columns) from my recycleView
+            GridLayoutManager recycleLayoutManager = new GridLayoutManager(view.getContext(), 2, GridLayoutManager.VERTICAL, false);
+            picturesRecycleView.setLayoutManager(recycleLayoutManager);
+            picturesRecycleView.setItemAnimator(new DefaultItemAnimator());
+
+            //Load the cards using the RecycleCardsAdapter into the picturesRecycleView
+            ArrayList<GalleryManager.PhotoStructure> photos = GalleryManager.getImages(view.getContext());
+            cardAdapter = new GalleryManager.RecycleCardsAdapter(view.getContext(), photos);
+            picturesRecycleView.setAdapter(cardAdapter);
+        }
+
+
+        /**
+         * Event when the main fragment change its hidden status.
+         * @param hidden true=is hidden, false=is not hidden.
+         */
+        @Override
+        public void onHiddenChanged(boolean hidden)
+        {
+            //I have to understand if the user has deleted an image
+            //If the answer is yes I have to remove the item from the recycleView
+            if(!hidden)
+            {
+                //When I delete something from the DetailsFragment using the method GalleryManager.deleteImage(photoInfos)
+                //I set in the GalleryManager an object containing the PhotoStructure to delete.
+                //Calling .getDeletedPhotoPosition() I check if the value is -1, if not I get the position of
+                //the PhotoStructure to delete
+
+                int deletedPosition=cardAdapter.getDeletedPhotoPosition();
+                if(deletedPosition!=-1)
+                {
+                    //have to remove the photo from the recycler
+                    cardAdapter.photosList.remove(deletedPosition);
+                    cardAdapter.notifyItemRemoved(deletedPosition);
+                }
+
+                //Restore the actionBar
+                ActionBar actionBar =((GalleryActivity)getActivity()).getSupportActionBar();
+                actionBar.setTitle(R.string.galleryFragmentTitle);
+                actionBar.setDisplayHomeAsUpEnabled(false);
+                actionBar.setHomeButtonEnabled(false);
+            }
+        }
+    }
+
+    /**
+     * Fragment of the detailed photo layout
+     * @author Romanello Stefano
+     */
+    public static class DetailFragment extends Fragment
```

Comment:
Which class use this class named DetailFragment? It seems that it's not launched by GalleryActivity or GalleryActivity.MainFragment. Is it right?
If yes, could a good idea be making a single class for this part of code to improve modularity and to make the code clearer than this?

Reply:
is used from the GalleryManager

Reply:
What is the advantage to include the implementation of plus than one public fragment class here?
There are other ways to organize the work, such as in packages or in modules if the scope is to isolate some specific classes from each others.